runtime/v2/runc/v2/service.go

```diff
@@ -690,14 +701,30 @@ func (s *service) getContainerPids(ctx context.Context, id string) ([]uint32, er
 	return pids, nil
 }
 
+type publishedEvent struct {
+	count int
+	evt   interface{}
+}
+
 func (s *service) forward(publisher events.Publisher) {
 	for e := range s.events {
-		ctx, cancel := context.WithTimeout(s.context, 5*time.Second)
-		err := publisher.Publish(ctx, runc.GetTopic(e), e)
-		cancel()
-		if err != nil {
-			logrus.WithError(err).Error("post event")
+		if e.count > maxEventRetry {
+			logrus.WithFields(logrus.Fields{
+				"topic": runc.GetTopic(e.evt),
+			}).Error("max retry count reached for event")
+			continue
+		}
+		ctx, cancel := context.WithTimeout(s.context, time.Duration(5*e.count)*time.Second)
+		if err := publisher.Publish(ctx, runc.GetTopic(e.evt), e.evt); err != nil {
+			e.count++
+			go func() {
+				// short pause before requeue of the event
+				time.Sleep(100 * time.Millisecond)
+				s.events <- e
```

Comment:
Event may be out of order. It is fine, because we have timestamp on event, but we may want to keep this in mind.

Reply:
Do we have any documented guarantees? Would it be useful if events were blocked on retry just for the same containerid?